### PR TITLE
Added tensor transforms jupyter notebook

### PR DESCRIPTION
**Pull request by vfdev-5**
Description:

- Updated docs and provided an example of scriptable transforms

### CODE REVIEW

File: examples/python/scriptable_transforms/inference.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        super().__init__()
+        self.resnet18 = resnet18(pretrained=True).eval()
+        self.transforms = nn.Sequential(
+            T.Resize(256),
+            T.CenterCrop(224),
+            T.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
+        )
+
+    def forward(self, x: torch.Tensor) -> int:
+        with torch.no_grad():
```
Comment (fmassa) [inline examples/python/scriptable_transforms/inference.py:21]: nit: I would probably use this as a context manager
```python
@torch.no_grad()
def forward(...)
```

File: examples/python/scriptable_transforms/inference.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            x = x.float() / 255.0
+            x = self.transforms(x)
+            y_pred = self.resnet18(x.unsqueeze(0))
+            return y_pred[0].argmax().item()
+
+
+if __name__ == "__main__":
+
+    path = "ILSVRC2012_val_00017789.JPEG"
+    img_tensor = read_image(path)
```
Comment (fmassa) [inline examples/python/scriptable_transforms/inference.py:31]: With https://github.com/pytorch/vision/pull/2728 `read_image` will also be torchscriptable, so we could even make this part of the model if we want

**Review by fmassa — COMMENTED** [review]
Thanks for the PR!

I have some early comments, which are nits

**Comment by vfdev-5** [comment]
ToDo: Improve inference example
- Inference on batch input (CPU and GPU)

File: examples/python/README.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# Python examples
+
+- [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/pytorch/vision/blob/master/examples/python/tensor_transforms.ipynb)
```
Comment (fmassa) [inline examples/python/README.md:3]: Nice! I didn't know about the badge!

File: examples/python/tensor_transforms.ipynb [hunk 4]
```diff
@@ -0,0 +1,611 @@
...
+        "id": "vjAC2mZnb4nz"
+      },
+      "source": [
+        "# Image transformations\n",
+        "\n",
+        "This notebook shows new features of torchvision image transformations:\n",
+        "- can accept torch tensor images as input\n",
+        "- can run on GPU\n",
+        "- can be torch jit scripted (except `Compose`, `RandomApply`, `RandomChoice`, `Lambda` transforms).\n",
+        "- can be applied on batch of tensors\n"
```
Comment (fmassa) [inline examples/python/tensor_transforms.ipynb:42]: I think it would be nice to give some more context here first.
Maybe mention that the transforms in torchvision have traditionally been PIL-centric, and as such they didn't support many things like
- multi-band images (with more than 3-4 channels)
- to be torchscripted together with your model for deployment
- support for GPU acceleration
- support for batched transformation (such as for videos)

And that now with a Tensor-first implementation (while maintaining PIL support for backwards-compatibility), we can achieve all of the above.

File: examples/python/tensor_transforms.ipynb [hunk 5]
```diff
@@ -0,0 +1,611 @@
...
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "Epp3hCy0b4oD"
+      },
+      "source": [
+        "!wget https://farm1.static.flickr.com/152/434505223_8d1890e1e2.jpg -q -Otest-image.jpg\n",
+        "!wget https://farm3.static.flickr.com/2142/1896267403_24939864ba.jpg -q -Otest-image2.jpg"
```
Comment (fmassa) [inline examples/python/tensor_transforms.ipynb:124]: I would probably use something like
```python
requests.get(url, stream=True).raw
```
as `wget` is not present by default on OSX nor Windows

File: examples/python/tensor_transforms.ipynb [hunk 6]
```diff
@@ -0,0 +1,611 @@
...
+        "outputId": "e5de5c73-e16d-4992-ebee-94c7ddf0bf54",
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 303
+        }
+      },
+      "source": [
+        "from torchvision.io.image import read_image\n",
+        "\n",
+        "tensor_image = read_image(\"test-image.jpg\")\n",
```
Comment (fmassa) [inline examples/python/tensor_transforms.ipynb:154]: I think it would be good to mention that torchvision now contains image reading functions for PNG and JPG images, with torchscript support, so that image decoding can be part of your model as well.

File: examples/python/tensor_transforms.ipynb [hunk 7]
```diff
@@ -0,0 +1,611 @@
...
+        "print(\"output tensor image info: \", out_image.shape, out_image.dtype)\n",
+        "\n",
+        "plt.imshow(out_image.numpy().transpose((1, 2, 0)))"
+      ],
+      "execution_count": 32,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            "Clipping input data to the valid range for imshow with RGB data ([0..1] for floats or [0..255] for integers).\n"
```
Comment (fmassa) [inline examples/python/tensor_transforms.ipynb:229]: Can we fix this warning on `imshow`? It's because of `Normalize` most probably

File: examples/python/tensor_transforms.ipynb [hunk 8]
```diff
@@ -0,0 +1,611 @@
...
...7Tm6njDgoXFFrW2Q1cp6sD22HAU1urQsCpmNbkteW4qaccxDPpjer0BfhASRXU8D4ZDizAMiaKoZO6x8Dwb21HowiLNcrwsx7ItijzHKvP3nmd8MMOaY5EksWkb5jhEUY0oimg0mxRpzuLiIvW6acM1HA7JDx/Gn5tC3Xsf9f378DyfcHGBIk7ozkwbJXDxBVhZAwYQ7TL3fnDzZqLuia/hI9f69Ea36DfYWb3x7yKDUYbONLFt03f6+GHEnj0z+K7NcLiFUgmO55nSYK+Otm2sIiBNY9J0hE4TlM5RymRlknSLtaUNFndr9u0/yLQfcvfuRZKZjK2lC2w/t8Xa+YI/OnkWz7vE9ev/fyAVkXy51BFIcY8E20T4BHwjq7UIttQZbFAJqwTxhLE4wAi8EIZKJkBWZ4ERT5fb16kqDgXAMwlZlv19jAUhXALCRCS4ALEyhBdgm8oFaVPRi62Xn4PyeLvL46+zU5W40FYcax+izsHyZjimAMft4ixm7PXnaB/LuHezxpnNJs8tBzy/kvDEF06yee4U0cwCOu1x4dwpNro9rn72Caz6DO9+/7fTai/wuSc/w4MnTrC4N+LKtmJh2kx3ZRO63Zz9oYWP2ilzyLTJFozHMBhoup2Yra0B3W7MjFvD8wzyME0TwtBnZqYByiDjanWF7ymGI+j3FbnjYFkW49EYx7FptdsMBwPDL+g6KEz3H8dx8X1jZfi+b9h3Oh32HzjA9vY2QRCwtrqKHfkox2Lu6F1Mz8+SJjlWEZJlKXFR0KhFFHWbzZVnmbnvjTj7TrDy5J/dUgkc++p38Ms/9ZNsr63cZIu7gbMv/VoDSUK+ZdGxbTy3YNAbYzVCPC+k2XLQWnF9aY3la5fR2iZzLZzAp9YMaEQNPEujioQiS9haHzAaDHCLGqHlocYpe6dmuHr1KqdP9+h2gX5KvPoJXgnafOcogV1U8FqpEWhSlQRPU6XKFjDCOODGnoNgBHmNKli4jhHMqYlt5oB9mLSilP9O0onVMVaAVZ5rA4MZmGQrnkT4iVXgYoS6yY18gJqKZViCfBGVm2JRZUeEayCY2OdyeQ0t2JvCjGpNTKREJal7Uf5dBLs0/nyNVlon6GbEFzZYeeIUafwCUWue/QeOoHsJz/71E2yePc8wzviG73kXWysZf/Kffo63vPvdLByuc3Ed9s/CpW3YGuZsbA45e
```
Comment (fmassa) [inline examples/python/tensor_transforms.ipynb:298]: Maybe we can re-use the code from the previous block?

Comment (fmassa) [inline examples/python/tensor_transforms.ipynb:365]: Maybe make the title something like
```
Scriptable transforms for easier deployment via torchscript
```

Comment (fmassa) [inline examples/python/tensor_transforms.ipynb:371]: I would also explain why we want to use `nn.Sequential` instead of `Compose` (BC and because `Compose` is simply a `Sequential`)

Comment (fmassa) [inline examples/python/tensor_transforms.ipynb:563]: I would say to leave this for a follow-up PR

**Review by fmassa — COMMENTED** [review]
Thanks a lot for the improvements, this is looking great!

I've made some more comments, let me know what you think

**Review by fmassa — COMMENTED** [review]
This is looking great, thanks!

I can't comment on the specific lines of the notebook because it's too large (due to the images) so GitHub won't show it. There are a few minor things that I think could be improved, but they can be added in a follow-up PR.

There are a few empty cells that could be cleaned up as well.

Also, is there a way of making the notebook a bit smaller? It's currently > 1MB, maybe cleaning up the images (by removing the output of all cells) could help there?

**Comment by vfdev-5** [comment]
@fmassa thanks for the review, I can address all that in this PR:
- remove images, empty cells and ToDo section
- replace all mentions of "torch jit script" by "torchscript"

**Review by fmassa — APPROVED** [review]
Thanks a lot!
